src/ec/packages/ec-component-radio/radio-group.html.twig

```diff
@@ -59,44 +56,48 @@
 
 {# Print the result #}
 
-<div
+<fieldset
   class="{{ _css_class }}"
-  role="radiogroup"
-  aria-labelledby="{{ label_id }}"
-  {{ _helper_id is not empty ? "aria-describedby=\"" ~ _helper_id ~ "\""}}
+  {{ _helper_id is not empty ? 'aria-describedby="' ~ _helper_id ~ '"'}}
   {{ _extra_attributes|raw }}
 >
 
   {%- if _label is not empty %}
-    <div
-      class="ecl-form-label{{ hide_label ? ' ecl-form-label--hidden'}}"
-      id="{{ _label_id }}"
+    <legend
+      class="ecl-form-label
+      {{- _invalid ? ' ecl-form-label--invalid' }}"
     >
       {%- block label _label -%}
-    </div>
+      {%- if _required -%}
+        <span class="ecl-form-label__required">{{ _required_text }}</span>
+      {%- else -%}
+        <span className="ecl-form-label__optional">{{ _optional_text }}</span>
+      {%- endif -%}
+    </legend>
   {% endif %}
 
   {% if _helper_text is not empty %}
-    <p
-      class="ecl-radio__help ecl-help-block"
-      {{ _helper_id is not empty ? "id=\"" ~ _helper_id ~ "\""}}
+    <div
+      class="ecl-help-block"
+      {{ _helper_id is not empty ? 'id="' ~ _helper_id ~ '"'}}
     >
       {%- block helper_text _helper_text -%}
-    </p>
+    </div>
   {% endif %}
 
   {% if _invalid and _invalid_text is not empty %}
-    <p class="ecl-radio__invalid ecl-feedback-message">
+    <p class="ecl-feedback-message">
```

Comment:
Didn't we say these help blocks are all wrapped in a div container? ;)

Reply:
Yes :) thanks for spotting this! done